x/programs/rust/examples/nft/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+//! A basic ERC-721 compatible contract.
+//! The program serves as a non-fungible token with the ability to mint, burn, and transfer.
+//! Only supports whole units with no decimal places.
+//!
+//! NOTE: The NFT must support the common NFT metadata format.
+//! This JSON encoded file provides all the necessary metadata about the NFT.
+use metadata::NFT;
+use wasmlanche_sdk::{program::Program, public, state_keys, types::Address};
+
+pub mod metadata;
+
+const NFT_NAME: &str = "My NFT";
+const NFT_SYMBOL: &str = "MNFT";
+const NFT_TOTAL_SUPPLY: u64 = 1;
+
+/// Initializes the NFT with all required metadata.
+/// This includes the name, symbol, image URI, owner, and total supply.
+/// Returns true if the initialization was successful.
+#[public]
+pub fn init(program: Program, owner: Address) -> bool {
+    // set token name
+    program
+        .state()
+        .store(StateKey::Name.to_vec(), &NFT_NAME.as_bytes())
+        .expect("failed to store nft name");
+
+    // set token symbol
+    program
+        .state()
+        .store(StateKey::Symbol.to_vec(), &NFT_SYMBOL.as_bytes())
+        .expect("failed to store nft symbol");
+
+    // Generate NFT metadata and persist to storage
```

Comment:
minor: all inline comments should start with Uppercase or lowercase please align on one :)